x/dkim/keeper/query_server.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkError "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/burnt-labs/xion/x/dkim/types"
+)
+
+var _ types.QueryServer = Querier{}
+
+type Querier struct {
+	Keeper
+}
+
+func NewQuerier(keeper Keeper) Querier {
+	return Querier{Keeper: keeper}
+}
+
+func (k Querier) Params(c context.Context, _ *types.QueryParamsRequest) (*types.QueryParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	p, err := k.Keeper.Params.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryParamsResponse{Params: &p}, nil
+}
+
+// DkimPubKeys implements types.QueryServer.
+func (k Querier) DkimPubKey(ctx context.Context, msg *types.QueryDkimPubKeyRequest) (*types.QueryDkimPubKeyResponse, error) {
+	dkimPubKey, err := k.OrmDB.DkimPubKeyTable().Get(ctx, msg.Domain, msg.Selector)
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryDkimPubKeyResponse{DkimPubkey: &types.DkimPubKey{
+		Domain:   dkimPubKey.Domain,
+		PubKey:   dkimPubKey.PubKey,
+		Selector: dkimPubKey.Selector,
+	}, PoseidonHash: []byte(dkimPubKey.PubKey)}, nil
+}
+
+// PoseidonHash implements types.QueryServer.
+func (k Querier) PoseidonHash(_ context.Context, msg *types.PoseidonHashRequest) (*types.PoseidonHashResponse, error) {
+	hash, err := types.ComputePoseidonHash(msg.PublicKey)
```

Comment:
This data is stored on the record. Exposing hash functions that users can hit for free, is a guaranteed spam vector, allowing people to force RPC nodes to constantly run hash functions. This value should be read from the record, like in the other query

Reply:
It is also possible this query doesn't need to exist, since all of the data is provided in the other query

Reply:
I thought you said we needed an endpoint people can call to compute the hash from the CLI